src/tasks/helpers/extensionDescriptorPaths.js

```diff
@@ -31,26 +33,45 @@ if (isSandboxLinked()) {
   );
 }
 
-// Remove duplicate paths.
-extensionJsonPaths = extensionJsonPaths.filter(
-  (value, index, self) => self.indexOf(value) === index
-);
-
-// Order the paths so that `extension.json` is last in the array. We do this because
-// we may use the sandbox when we develop the core extension. We will be in a situation where
-// the core extension has sandbox as a dependency. And the sandbox has the core extension
-// as a dependency (Core -> Sandbox -> Core). We want to load the core extension that has the
-// sandbox as a dependency (meaning we want the extension with extension.json path to always win).
-extensionJsonPaths.sort((a, b) => {
-  if (a === EXTENSION_DESCRIPTOR_FILENAME) {
-    return 1;
+function getExtensionPlatform(filePath) {
+  const fileContents = JSON.parse(fs.readFileSync(path.resolve(filePath)));
+  if (!fileContents.platform) {
+    throw new Error(
+      `Unable to detect the platform the development
+      extension is targeting for path "${filePath}"`
+    );
   }
+  return fileContents.platform;
+}
 
-  if (b === EXTENSION_DESCRIPTOR_FILENAME) {
-    return -1;
-  }
+// if there's only 1 path, skip over this logic
+if (extensionJsonPaths.length > 1) {
+  // Order the paths so that `extension.json` is last in the array. We do this because
+  // we may use the sandbox when we develop the core extension. We will be in a situation where
+  // the core extension has sandbox as a dependency. And the sandbox has the core extension
+  // as a dependency (Core -> Sandbox -> Core). We want to load the core extension that has the
+  // sandbox as a dependency (meaning we want the extension with extension.json path to always win).
+  extensionJsonPaths.sort((a, b) => {
+    if (a === EXTENSION_DESCRIPTOR_FILENAME) {
+      return 1;
+    }
 
-  return 0;
-});
+    if (b === EXTENSION_DESCRIPTOR_FILENAME) {
+      return -1;
+    }
+
+    return 0;
+  });
+
+  const extensionPlatform = getExtensionPlatform(extensionJsonPaths[extensionJsonPaths.length - 1]);
```

Comment:
I would inject the platform of the extension that runs the sandbox when we call this module. It is not really clear why you read it from the last entry of the array.

Reply:
Would you change the contract of the module? Right now we just `require` it in different files.

Reply:
I would make it a function and memorize it. I don't have a strong opinion. If you like it like this, I can approve it. But I am thinking that if we change the way we sort the descriptors, there is a high chance that this line won't be changed.

Reply:
The problem is that we don't have tests for sandbox and it is very fragile.

Reply:
I've got an idea! I'll push it soon, I just want to track down that data element syntax problem you were describing the other day.